packages/lodestar/src/chain/errors/attestationError.ts

```diff
@@ -0,0 +1,199 @@
+import {CommitteeIndex, Epoch, Slot, ValidatorIndex} from "@chainsafe/lodestar-types";
+
+export enum AttestationErrorCode {
+  /**
+   * The attestation is from a slot that is later than the current slot (with respect to the gossip clock disparity).
+   */
+  ERR_FUTURE_SLOT = "ERR_FUTURE_SLOT",
+  /**
+   * The attestation is from a slot that is prior to the earliest permissible slot
+   * (with respect to the gossip clock disparity).
+   */
+  ERR_PAST_SLOT = "ERR_PAST_SLOT",
```

Comment:
do we actually need to throw different errors for past and future slots?  the spec rule is worded to where a single IGNORE is thrown when the slot is not within the allowed range.  maybe we combine ERR_FUTURE_SLOT and ERR_PAST_SLOT into one error and call it ERR_SLOT_OUT_OF_RANGE or something like that?

Here's the rule from the spec, btw:
```    [IGNORE] attestation.data.slot is within the last ATTESTATION_PROPAGATION_SLOT_RANGE slots (within a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. attestation.data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= current_slot >= attestation.data.slot (a client MAY queue future attestations for processing at the appropriate slot).```

Reply:
These errors aren't necessarily supposed to be 1 for 1 with gossip validation rules.
They're meant to enumerate meaningfully distinct errors that can happen in the course of processing a block.
Likely, when we're incorporating them into Lodestar further, some cases will be deleted, some will be added.
This list is taken mainly from Lighthouse's block errors.

IMO we should save this discussion for a future PR, when we incorporate these errors into validation functions. At that point it will be more clear whether past vs future slot a meaningful distinction to make.